sw/device/lib/dif/dif_rv_core_ibex.h

```diff
@@ -159,6 +159,69 @@ typedef enum dif_rv_core_ibex_nmi_source {
   kDifRvCoreIbexNmiSourceAll = 0x3,
 } dif_rv_core_ibex_nmi_source_t;
 
+typedef struct dif_rv_core_ibex_crash_dump_state {
+  /**
+   * The last exception address.
+   */
+  uint32_t mtval;
+
+  /**
+   * The last exception PC.
+   */
+  uint32_t mpec;
+
+  /**
+   * The last data access address.
+   */
+  uint32_t mdaa;
+
+  /**
+   * The next PC.
+   */
+  uint32_t mnpc;
+
+  /**
+   * The current PC.
+   */
+  uint32_t mcpc;
+} dif_rv_core_ibex_crash_dump_state_t;
+
+typedef struct dif_rv_core_ibex_previous_crash_dump_state {
+  /**
+   * The exception address for the previous crash.
+   */
+  uint32_t mtval;
+
+  /**
+   * The last exception PC for the previous crash.
+   */
+  uint32_t mpec;
+} dif_rv_core_ibex_previous_crash_dump_state_t;
+
+/**
+ * Under normal circumstances, only the current crash dump state is valid. When
+ * the CPU encounters a double fault, the current crash dump is moved to
+ * previous, and the new crash dump is shown in current.
+ */
+typedef struct dif_rv_core_ibex_crash_dump_info {
+  /**
+   * The crash dump state for the current execution.
+   */
+  dif_rv_core_ibex_crash_dump_state_t fault_stage;
+
+  /**
+   * The crash dump state for the previous execution. I will only contain valid
```

Comment:
does it make sense to have another field to indicate whether the prev state is valid?

Reply:
i'm imagining if someone uses this to debug, they won't know whether a double fault occurred without looking at the "valid" field, because on double fault, the processor actually stops execution, so this is the only hint.